protocol/msgpack/base_iterator.go

```diff
@@ -39,13 +39,18 @@ var (
 
 // baseIterator is the base iterator that provides common decoding APIs
 type baseIterator struct {
-	decoder   *msgpack.Decoder // internal decoder that does the actual decoding
-	decodeErr error            // error encountered during decoding
+	decoder            *msgpack.Decoder    // internal decoder that does the actual decoding
+	policyDecompressor policy.Decompressor // decompressor used to decompress policies
+	decodeErr          error               // error encountered during decoding
 }
 
-func newBaseIterator(reader io.Reader) iteratorBase {
+func newBaseIterator(reader io.Reader, opts PolicyDecompressionOptions) iteratorBase {
```

Comment:
Sounds good, calling it a `BaseIteratorOptions` to be more consistent with the encoder.